.github/workflows/build.yml

```diff
@@ -0,0 +1,46 @@
+name: build
+
+on:
+  push:
+    branches: ["main"]
+  pull_request:
+  workflow_dispatch:
+  
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    permissions:
+      # required by aws-actions/configure-aws-credentials
+      id-token: write
+      contents: read
+    steps:
+    - name: run number with offset
+      env:
+        NUM: ${{ github.run_number }}
+      run: echo "GITHUB_RUN_NUMBER_WITH_OFFSET=$((10000+$NUM))" >> $GITHUB_ENV
+    - uses: actions/checkout@v3
+    - name: Setup SBT environment
```

Comment:
I would suggest using [`setup-java`](https://github.com/actions/setup-java) instead of this third-party action. This is supported by GitHub, [promoted by `sbt`](https://www.scala-sbt.org/1.x/docs/GitHub-Actions-with-sbt.html) and used elsewhere ([e.g. AMIable](https://github.com/guardian/amiable/blob/c94c4b4ea51f85bff1eed5d96fe0bb55799c3df8/.github/workflows/ci.yml#L37-L43)). IIUC it's going to install the `sbt` version specified in [`build.properties`](https://github.com/guardian/elastic-search-monitor/blob/main/project/build.properties), which also means that we only need to manage this version in one place.

Reply:
That was a very very helpful comment, thank you!! Will use this going forward